### PR TITLE
htmlHint

### PR DESCRIPTION
**Pull request by delphiactual**
htmlHint with sane defaults

Found some extra and/or missing `<divs>`, `</divs>`, `<ng-repeat=`, `<`, `>`

http://htmlhint.com/

### CODE REVIEW

File: src/scripts/item-review/item-review.html [hunk 1]
```diff
@@ -32,11 +32,11 @@
   <div ng-if="$ctrl.procon" class="community-review--procon">
     <div>
       <div translate="DtrReview.Pros"></div>
-      <div ng-repeat="pro in $ctrl.aggregate.pros" ng-bind='pro'></div>
+      <div ng-repeat="pro in $ctrl.aggregate.pros" ng-bind="pro"></div>
     </div>
     <div>
       <div translate="DtrReview.Cons"></div>
-      <div ng-repeat="pro in $ctrl.aggregate.cons" ng-bind='pro'></div>
+      <div ng-repeat="pro in $ctrl.aggregate.cons" ng-bind="pro"></div>
```
Comment (delphiactual) [inline src/scripts/item-review/item-review.html]: @48klocs should this be `<div ng-repeat="con in $ctrl.aggregate.cons" ng-bind="con"></div>`

Comment (kyleshay) [inline src/scripts/item-review/item-review.html]: yeah, should be. though `$ctrl.procon` is hardcoded false currently anyway.

File: src/scripts/loadout-builder/loadout-builder.html [hunk 2]
```diff
@@ -7,10 +7,10 @@ <h1 translate="LB"></h1>
   <dim-loadout></dim-loadout>
 
   <p>
-    <span class='dim-button' ng-click="vm.showGear=!vm.showGear" translate-values="{ class: vm.i18nClassNames[vm.active] }" translate>{{!vm.showGear ? 'LB.ShowGear' : 'LB.HideGear'}}</span>
+    <span class="dim-button'" ng-click="vm.showGear=!vm.showGear" translate-values="{ class: vm.i18nClassNames[vm.active] }" translate>{{!vm.showGear ? 'LB.ShowGear' : 'LB.HideGear'}}</span>
```
Comment (kyleshay) [inline src/scripts/loadout-builder/loadout-builder.html]: > "dim-button'"
trailing `'`

File: src/scripts/itemDiscuss/dimItemDiscuss.directive.html [hunk 3]
```diff
@@ -79,4 +79,4 @@
             </div>
         </div>
     </div>
-</div>
\ No newline at end of file
+</div>
```
Comment (kyleshay) [inline src/scripts/itemDiscuss/dimItemDiscuss.directive.html]: we should be consistent with end of line new lines between js + css + html

the `.editorconfig` file should set that for your depending on your IDE

Comment (delphiactual) [inline src/scripts/itemDiscuss/dimItemDiscuss.directive.html]: I'll check, I just switched from atom over to VS Code today, so maybe I forgot to set up editorconfig

Comment (kyleshay) [inline src/scripts/itemDiscuss/dimItemDiscuss.directive.html]: ok, no prob!

**Comment by delphiactual** [comment]
Something about `body` does not like imported templates...

File: src/scripts/move-popup/dimTalentGrid.directive.html [hunk 4]
```diff
@@ -1,9 +1,9 @@
...
- <g ng-attr-transform="scale({{vm.scaleFactor}})">
-  <g class="talent-node" ng-attr-transform="translate({{(node.column - vm.hiddenColumns) * (vm.totalNodeSize)}},{{node.row * (vm.totalNodeSize)}})" ng-repeat="node in vm.talentGrid.nodes | talentGridNodes:vm.hiddenColumns track by $index" ng-class="{ 'talent-node-activated': node.activated, 'talent-node-showxp': (!node.activated && node.xpRequired), 'talent-node-default': (node.activated && !node.xpRequired && !node.exclusiveInColumn) }" ng-click="node.hash == 1270552711 ? vm.infuse({'$event': $event}) : true">
-    <circle r="16" cx="-17" cy="17" transform="rotate(-90)" class="talent-node-xp" ng-attr-stroke-width="{{node.xp ? 2 : 0}}" ng-attr-stroke-dasharray="{{100.0 * node.xp / node.xpRequired}} 100" />
+<svg preserveaspectratio="xMaxYMin meet" svg-bind-viewbox="0 0 {{ (vm.numColumns * vm.totalNodeSize - vm.nodePadding) * vm.scaleFactor }} {{ (vm.numRows * vm.totalNodeSize - vm.nodePadding) * vm.scaleFactor + 1 }}" class="talent-grid" ng-attr-height="{{ (vm.numRows * vm.totalNodeSize - vm.nodePadding) * vm.scaleFactor }}" ng-attr-width="{{ (vm.numColumns * vm.totalNodeSize - vm.nodePadding) * vm.scaleFactor }}">
```
Comment (bhollis) [inline src/scripts/move-popup/dimTalentGrid.directive.html]: SVG is particular about the case of attributes - you probably want to keep it `preserveAspectRatio`.

Comment (delphiactual) [inline src/scripts/move-popup/dimTalentGrid.directive.html]: fixed

**Comment by delphiactual** [comment]
@bhollis I do not understand how to get this to work from the command line, this might only be possible by using plugins within IDEs.

**Comment by kyleshay** [comment]
change line 13 in package.json to be `"lint": "htmlhint;eslint .",` then try running `npm run lint`

**Comment by kyleshay** [comment]
we could instead run htmlhint just in src by doing something like: `htmlhint src`

**Comment by kyleshay** [comment]
we should also just remove `/app`

**Comment by delphiactual** [comment]
in this PR or a separate?

**Comment by kyleshay** [comment]
let's do it in another

**Comment by bhollis** [comment]
Ready to merge?

**Comment by delphiactual** [comment]
🚢

**Pull request merged**